marklogic-data-hub-central/ui/src/components/entities/create-edit-step/create-edit-step.tsx

```diff
@@ -213,7 +213,13 @@ const CreateEditStep: React.FC<Props>  = (props) => {
   };
 
   const handleSearch = async (value: any) => {
-    let database = props.editStepArtifactObject["targetDatabase"] === "data-hub-FINAL" ? "final" : "staging";
+    let database: string = "";
+    if (!props.isEditing && props.stepType === StepType.Matching || props.stepType === StepType.Merging) {
+      database = "final";
+    } else if (props.isEditing) {
+      database = props.editStepArtifactObject["targetDatabase"] === "data-hub-FINAL" ? "final" : "staging";
```

Comment:
I believe this should be based on the sourceDatabase value, since that is what is queried and this search is used to populate the source collection value.

Reply:
Updated edit database to sourceDatabase.